.github/workflows/go-lint.yml

```diff
@@ -0,0 +1,28 @@
+name: golangci-lint
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+
+env:
+  GOLANGCI_LINT_VERSION: v1.61.0
+jobs:
+  lint-go:
+    runs-on: ubuntu-latest
```

Comment:
Nit: Similar to the other workflow, I'd add a human readable name for consistency

```suggestion
  lint-go:
    name: Lint Go
    runs-on: ubuntu-latest
```